### ISSUE TITLE
missing-type-doc / W9016 doesn't work for class Methods

### ISSUE THREAD

**Issue by skanderboudawara** [pylint-dev/pylint#8997]
### Bug description

We want to type hint all our project so we used : missing-type-doc / W9016

It function well when declaring a function the detecton works fine
but when the function is inside a class (so It's a method) there is no detection.

`def sum(a,b)` -> will raise the error that missing-type-doc / W9016

```
class Math():
     def sum(a,b)
```

There will be no detection

### Configuration

```ini
[MAIN]
# Files or directories matching the regex patterns are skipped. The regex
# matches against base names, not paths.
ignore-patterns=(.)*_test\.py,test_(.)*\.py,pipeline\.py,setup\.py,__init__\.py


# List of plugins (as comma separated values of python modules names) to load,
# usually to register additional checkers.
load-plugins=
    pylint.extensions.docparams,
    pylint.extensions.emptystring,
    pylint.extensions.comparetozero,
    pylint.extensions.docstyle,
    pylint.extensions.empty_comment,
    pylint.extensions.check_elif

# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
# number of processors available to use.
jobs=0

# When enabled, pylint would attempt to guess common misconfiguration and emit
# user-friendly hints instead of false-positive error messages.

# Activate the evaluation score.
score=no
fail-under=10

# Pickle collected data for later comparisons.
persistent=yes

# When enabled, pylint would attempt to guess common misconfiguration and 
# emit user-friendly hints instead of false-positive error messages.
suggestion-mode=yes

# To disable modules
disable=
    similarities,
    design,
    format,
    spelling,

[BASIC]
# Good variable names which should always be accepted, separated by a comma
good-names = 
    _,
    i,
    j,
    k,
    df,
    w,
    e,
    force_eqNull,
    sqlCtx,
    w1,
    w2,
    C,
    dd,
    xwb,
    sa,
    pn,
    lr,
    tc,
    v

# Bad variable names which should always be refused, separated by a comma
bad-names=
    foo,
    bar,
    baz,
    toto,
    tutu,
    tata

# Minimum line length for functions/classes that require docstrings, shorter ones are exempt.
docstring-min-length=1
    
[MESSAGES CONTROL]
# Disable the message, report, category or checker with the given id(s). You
# can either give multiple identifiers separated by comma (,) or put this
# option multiple times (only on the command line, not in the configuration
# file where it should appear only once).You can also use "--disable=all" to
# disable everything first and then re-enable specific checks. For example, if
# you want to run only the similarities checker, you can use "--disable=all
# --enable=similarities". If you want to run only the classes checker, but have
# no Warning level messages displayed, use"--disable=all --enable=classes
# --disable=W"
disable =
    line-too-long,
    attribute-defined-outside-init,
    no-self-argument,
    useless-parent-delegation,
    too-many-lines,
    unused-private-member,
    too-many-public-methods,
    too-many-instance-attributes,
    too-few-public-methods,
    too-many-locals,
    logging-fstring-interpolation,
    missing-raises-doc,
    redundant-returns-doc,
    too-many-return-statements,
    too-many-branches,
    too-many-statements,
    too-many-arguments,
    useless-option-value,
    no-self-use,
    unknown-option-value,
    R6301,
    useless-super-delegation,
    super-init-not-called,
    redundant-keyword-arg,
    docstring-first-line-empty,
    RP0801,
    R0801,
    unnecessary-lambda-assignment,
    no-member,
    unnecessary-pass


[STRING]
# This flag controls whether inconsistent-quotes generates a warning when the
# character used as a quote delimiter is used inconsistently within a module.
check-quote-consistency=yes


[TYPECHECK]
# List of module names for which member attributes should not be checked 
# (useful for modules/projects where namespaces are manipulated during runtime and 
# thus existing member attributes cannot be deduced by static analysis). 
# It supports qualified module names, as well as Unix pattern matching.
ignored-modules=
    transforms,
    transforms.api._decorators,
    transforms.expectations,
    expectations,
    pyspark,
    pyspark.sql,
    pyspark.sql.types,
    pyspark.sql.functions,
    pyspark.context,
    pandas,
    pytest,
    transforms.api,



[FORMAT]
max-line-length=400
max-module-lines=30000
```

### Command used

```shell
-
```

### Pylint output

```shell
-
```

### Expected behavior

To do the same detection for class Methods

### Pylint version

```shell
pylint=2.17.5
```

### OS / Environment

Palantir

**Comment by Pierre-Sassoulas** [comment]
Did you mean to close ? Is the issue fixed on your side ?

**Comment by fksontini** [comment]
Palantir ?
